src/discord/discord.service.ts

```diff
@@ -75,52 +91,15 @@ export class DiscordService implements OnModuleInit {
     }
   }
 
-  onInteractionCreate = (interaction: Interaction) => {
-    this.processInteraction(interaction).catch((error) => {
-      console.error('failed to process interaction', error);
-    });
-  };
-
   async processInteraction(interaction: Interaction) {
-    //TODO Can we extract this to a new file? Maybe echo.service.ts or echo.module.ts etc?
     if (!interaction.isChatInputCommand()) return;
 
     const { commandName } = interaction;
 
-    if (commandName === 'ping') {
-      await interaction.reply('Pong!');
-    }
-
-    if (commandName === 'echo') {
-      const message = interaction.options.getString('message', true);
-      await interaction.reply(message);
-    }
-
-    if (commandName === 'randomcabin') {
-      const checkIn = interaction.options.getString('check-in', true);
-      const checkOut = interaction.options.getString('check-out', true);
-      await interaction.reply('Not quite there yet, but working on it!');
-    }
-  }
-
-  async sendMessage(channelId: string, embed: EmbedBuilder) {
-    const channel = await this.client.channels.fetch(channelId);
-    if (!channel || !channel.isTextBased()) {
-      console.log('channel not found');
-      return false;
-    }
+    const path = `${__dirname}/../slashCommand/commands/${commandName}.command.js`;
```

Comment:
This logic belongs in the slash command module as we are reaching into its internals here. I would move this logic there and pass in the interaction.

The reason for this is that we are trying to make isolated modules where we can change anything internally and as long as the external API doesn't change then no other modules will need to be updated.

Reply:
I moved the code accordingly and pass the command name. This caused a circular dependency between discord service/module and slash command service/module that I resolved with forwardRef.

Reply:
I think in the future we can use register a call back with the discord service from the slash command service to handle this but your approach is good for now.